docs/proposals/templates.md

```diff
@@ -0,0 +1,550 @@
+
+This desire for simplicity also intentionally excludes template composability/embedding as a supported use case.  
+
+Allowing templates to reference other templates presents versioning+consistency challenges along with making the template 
+no longer a self-contained portable object.  Scenarios necessitating multiple templates can be handled in one of several 
+alternate ways:
+
+* Explicitly constructing a new template that merges the existing templates (tooling can easily be constructed to perform this 
+  operation since the templates are first class api objects).
+* Manually instantiating each template and utilizing [service linking](https://github.com/kubernetes/kubernetes/pull/17543) to share 
+  any necessary configuration data.
+
+This document will also refrain from proposing server APIs or client implementations.  This has been a point of debate, and it makes
+more sense to focus on the template/parameter specification/syntax than to worry about the tooling that will process or manage the
+template objects.  However since there is a desire to at least be able to support a server side implementation, this proposal
+does assume the specification will be k8s API friendly.
+
+## Desired characteristics
+
+* Fully k8s object json-compliant syntax.  This allows server side apis that align with existing k8s apis to be constructed 
+  which consume templates and existing k8s tooling to work with them.  It also allows for api versioning/migration to be managed by
+  the existing k8s codec scheme rather than having to define/introduce a new syntax evolution mechanism.
+  * (Even if they are not part of the k8s core, it would still be good if a server side template processing+managing api supplied 
+    as an ApiGroup consumed the same k8s object schema as the peer k8s apis rather than introducing a new one)
+* Self-contained parameter definitions.  This allows a template to be a portable object which includes metadata that describe
+  the inputs it expects, making it easy to wrapper a user interface around the parameterization flow.
+* Support for non-string parameters.  K8S objects define fields of type string, int, bool, and byte[](base64).  The parameter
+  scheme should support all of those types for substitution.
+  * complex types (struct/map/list) can be defined in terms of the available primitives, so it's preferred to avoid the complexity
+    of allowing for full complex-type substitution.
+* Parameter metadata.  Parameters should include at a minimum, information describing the purpose of the parameter, whether it is 
+  required/optional, and a default/suggested value.  Type information could also be required to resolve the non-string substitution
+  requirement.
+* Template metadata.  Templates should be able to include metadata describing their purpose or links to further documentation and 
+  versioning information.
+
+
+## Proposed Implementation
```

Comment:
This section should be split out as a separate PR, per discussion in the SIG, so that this PR deals only with requirements.